File: travelmate-userapp/src/app/offers/model/Offer.ts
import {Airplane} from "../../models/Airplane";
import {Airline} from "../../models/Airline";
import {Airport} from "../../models/Airport";

export interface Offer {
  id: string;
  airplane: Airplane;
  airline: Airline;
  departureAirport: Airport;
  arrivalAirport: Airport;
  flightNumber: string;
  departureTime: string;
  arrivalTime: string;
  basePrice: number;
  availableEconomySeats: number;
  availableBusinessSeats: number;
  availableFirstClassSeats: number;
  createdAt: string;
}

File: TravelMateBackend/Models/Entities/Offers/Offer.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using TravelMateBackend.Models.Entities;

namespace Models.Entities.Offers
{
    public class Offer
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public Airplane Airplane { get; set; }
        public Airline Airline { get; set; }
        public Airport DepartureAirport { get; set; }
        public Airport ArrivalAirport { get; set; }
        public string FlightNumber { get; set; }
        public DateTime DepartureTime { get; set; }
        public DateTime ArrivalTime { get; set; }
        public decimal BasePrice { get; set; }
        public int AvailableEconomySeats { get; set; }
        public int AvailableBusinessSeats { get; set; }
        public int AvailableFirstClassSeats { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}
